smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/httpResponse/bindingTraits/HttpResponseTraitWithoutHttpPayload.kt

```diff
@@ -133,4 +139,58 @@ class HttpResponseTraitWithoutHttpPayload(
     }
 
     private val path: String = "properties.".takeIf { outputShape.hasTrait<ErrorTrait>() } ?: ""
+
+    private fun writeInitialResponseMembers(initialResponseMembers: Set<HttpBindingDescriptor>) {
+        writer.apply {
+            write("if let initialDataWithoutHttp = await messageDecoder.awaitInitialResponse() {")
+            indent()
+            write("let decoder = JSONDecoder()")
+            write("do {")
+            indent()
+            write("let response = try decoder.decode([String: String].self, from: initialDataWithoutHttp)")
+
+            initialResponseMembers.forEach { responseMember ->
+                val responseMemberName = ctx.symbolProvider.toMemberName(responseMember.member)
+                write("self.$responseMemberName = response[\"$responseMemberName\"].map { value in KinesisClientTypes.Tag(value: value) }")
+            }
+
+            dedent()
+            write("} catch {")
+            indent()
+            write("print(\"Error decoding JSON: \\(error)\")")
+
+            initialResponseMembers.forEach { responseMember ->
+                val responseMemberName = ctx.symbolProvider.toMemberName(responseMember.member)
+                write("self.$responseMemberName = nil")
+            }
+
+            dedent()
+            write("}")
+            dedent()
+            write("} else {")
+            indent()
+
+            initialResponseMembers.forEach { responseMember ->
+                val responseMemberName = ctx.symbolProvider.toMemberName(responseMember.member)
+                write("self.$responseMemberName = nil")
+            }
+
+            dedent()
+            write("}")
```

Comment:
Make use of utility methods on SwiftWriter to keep codegen more readable + keep style uniform.
E.g., instead of manually indenting and dedenting, you can use `openBlock()` and `closeBlock()`.

```
writer.apply {
     openBlock("if let initialDataWithoutHttp = await messageDecoder.awaitIniitialResponse() {", "} else {") {
          // In if-block
          write("let decoder = JSONDecoder()")
          openBlock("do {", "} catch {") {
               // In do-block
               write("let response = try decoder.decode([String: String].self, from: initialDataWithoutHttp)")
               initialResponseMembers.forEach { responseMember ->
                    val responseMemberName = ctx.symbolProvider.toMemberName(responseMember.member)
                    write("self.$responseMemberName = response[\"$responseMemberName\"].map { 
                         value in KinesisClientTypes.Tag(value: value) 
                    }")
               }
          }
          // In catch-block
          write("print(\"Error decoding JSON: \\(error)\")")
          initialResponseMembers.forEach { responseMember ->
               val responseMemberName = ctx.symbolProvider.toMemberName(responseMember.member)
               write("self.$responseMemberName = nil")
          }
          // Close do-catch block
          closeBlock("}")
     }
     // In outer else-block
     initialResponseMembers.forEach { responseMember ->
                val responseMemberName = ctx.symbolProvider.toMemberName(responseMember.member)
                write("self.$responseMemberName = nil")
     }
     // Close if-else block
     closeBlock("}")
}
```

Reply:
Tried this out and no matter the configuration openBlock / closeBlock seem to produce spacing issues. Reverting this change for now.

Reply:
Discussed offline and decided on adding a helper method to code-generator for this use case in a separate PR.